### PR TITLE
225-allow-group-cid-offer

### PR DESCRIPTION
**Pull request by zenyuk**
https://consensyssoftware.atlassian.net/browse/FSR-225

### CODE REVIEW

File: pkg/register/register.go [hunk 1]
```diff
@@ -36,23 +36,24 @@ type RegisteredNode interface {
...
+	NodeID                 string `json:"nodeId"`
+	Address                string `json:"address"`
+	RootSigningKey         string `json:"rootSigningKey"`
+	SigningKey             string `json:"signingKey"`
+	RegionCode             string `json:"regionCode"`
+	NetworkInfoGateway     string `json:"networkInfoGateway"`
+	NetworkInfoProvider    string `json:"networkInfoProvider"`
+	NetworkInfoClient      string `json:"networkInfoClient"`
+	NetworkInfoAdmin       string `json:"networkInfoAdmin"`
+	GroupCIDOfferSupported bool   `json:"groupCIDOfferSupported""`
```
Comment (wcgcyx) [inline pkg/register/register.go:48]: We probably don't need the group cid offer support inside register

File: pkg/fcrmessages/gateway_list_dht_offer_request.go [hunk 2]
```diff
@@ -26,13 +26,14 @@ import (
...
-	MerkleRoot         string                       `json:"merkle_root"`
-	MerkleProof        fcrmerkletree.FCRMerkleProof `json:"merkle_proof"`
+	GatewayID            nodeid.NodeID                `json:"gateway_id"`
+	CIDMin               cid.ContentID                `json:"cid_min"`
+	CIDMax               cid.ContentID                `json:"cid_max"`
+	BlockHash            string                       `json:"block_hash"`
+	TransactionReceipt   string                       `json:"transaction_receipt"`
+	MerkleRoot           string                       `json:"merkle_root"`
+	MerkleProof          fcrmerkletree.FCRMerkleProof `json:"merkle_proof"`
+	PublishGroupCIDOffer bool                         `json:"publish_group_cid_offer"`
```
Comment (zenyuk) [inline pkg/fcrmessages/gateway_list_dht_offer_request.go:36]: remove PublishGroupCIDOffer

File: pkg/fcrmessages/gateway_list_dht_offer_request.go [hunk 3]
```diff
@@ -44,15 +45,17 @@ func EncodeGatewayListDHTOfferRequest(
 	transactionReceipt string,
 	merkleRoot string,
 	merkleProof *fcrmerkletree.FCRMerkleProof,
+	publishGroupCIDOffer bool,
```
Comment (zenyuk) [inline pkg/fcrmessages/gateway_list_dht_offer_request.go:48]: same as above

File: pkg/fcrmessages/gateway_list_dht_offer_request.go [hunk 4]
```diff
@@ -44,15 +45,17 @@ func EncodeGatewayListDHTOfferRequest(
...
-		MerkleRoot:         merkleRoot,
-		MerkleProof:        *merkleProof,
+		GatewayID:            *gatewayID,
+		CIDMin:               *cidMin,
+		CIDMax:               *cidMax,
+		BlockHash:            blockHash,
+		TransactionReceipt:   transactionReceipt,
+		MerkleRoot:           merkleRoot,
+		MerkleProof:          *merkleProof,
+		PublishGroupCIDOffer: publishGroupCIDOffer,
```
Comment (zenyuk) [inline pkg/fcrmessages/gateway_list_dht_offer_request.go:58]: same

File: pkg/fcrmessages/gateway_list_dht_offer_request_test.go [hunk 5]
```diff
@@ -57,14 +59,14 @@ func TestDecodeGatewayListDHTOfferRequest(t *testing.T) {
 	mockMerkleProof, _ := merkleTree.GenerateMerkleProof(mockCidMin)
 
 	validMsg := &FCRMessage{
-		messageType:200,
-		protocolVersion:1,
-		protocolSupported:[]int32{1, 1},
-		messageBody:[]byte(`{"gateway_id":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEI=","cid_min":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAE=","cid_max":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAI=","block_hash":"bafy2bzaceabhr6taytcdntpr4poz43dhf3l5jml6z43e5sdv4gdlgsujsg2ze","transaction_receipt":"bafy2bzacecz3sy5ar4rg73ri5cq3tndmwn4vnxgzt4bw4cpig7q25af6y5cnc","merkle_root":"c3c3a46684c07d12a9c238787df3049a6f258e7af203e5ddb66a8bd66637e108","merkle_proof":"AAAAMFsiQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBRT0iXQAAAANbMV0="}`), 
-		signature:"",
+		messageType:       200,
```
Comment (zenyuk) [inline pkg/fcrmessages/gateway_list_dht_offer_request_test.go:62]: update test, to remove the PublishGroupCIDOffer

File: pkg/fcrmessages/gatewayadmin_update_group_cid_offer_support_request.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/ConsenSys/fc-retrieval-common/pkg/nodeid"
+)
+
+type updateGatewayGroupCIDOfferSupportRequest struct {
+	GatewayID              nodeid.NodeID `json:"gateway_id"`
+	GroupCIDOfferSupported bool          `json:"group_cid_offer_supported"`
```
Comment (zenyuk) [inline pkg/fcrmessages/gatewayadmin_update_group_cid_offer_support_request.go:27]: additional param: ProviderID
so Gateway supports Group CID Offer not for all, but per Provider

**Review by wcgcyx — APPROVED** [review]
LGTM

**Pull request merged**